lib/deep-required/index.ts

```diff
@@ -1,4 +1,6 @@
+import { AnyArray } from "../any-array";
 import { Builtin } from "../built-in";
+import { Tuple } from "../tuple";
 
 export type DeepRequired<Type> = Type extends Error
```

Comment:
You still lack the tests for the new changes, please have a look at other Deep-utility types and add tests for:

- readonly arrays
- normal arrays
- tuples

If there're already in place, you can add only missing ones

Reply:
Yep - mentioned this in the issue. Will be sure to add more tests.

Reply:
Some of these were already in place, but I believe I've covered the missing ones. PTAL